### DISCUSSION TITLE
How to move from Arduino Due to ESP32 ?

### DISCUSSION THREAD

**Discussion by rroblik** [fredlcore/BSB-LAN#698] [Q&A] [answered]
Dear all

I have the official BSB LAN board from @fredlcore since... early 2021 (!) plugged into an Arduino Due. 

I would like to know how to move from Arduino to a ESP32 (and what "accessory I need to buy) ? A little lack of information on the documentation. 

The target board
![image](https://github.com/user-attachments/assets/97485181-741d-4a0e-86af-f569d537afac)

The potential accessory I think
![image](https://github.com/user-attachments/assets/8cdcac44-b6b9-49d6-8351-d45dd51aff4d)

> Therefore, you can re-use the Arduino Due board on an ESP32 microcontroller (if you connect the correct pins with wires), but you cannot use an adapter for the ESP32 on an Arduino Due.

Maybe that can help somebody else also !

have a nice day and merry Christmas 🎅🏽

**Comment by fredlcore** [comment, marked as answer]
When I thinned out the manual, this was one thing I dropped because it's a rather unusual move to make, as most of the functionality is the same between Due and ESP32.
But if for some reason you want to move (for example to use WiFi instead of Ethernet), then all you need to do is connect four lines between the adapter and the ESP32: RX1, TX1, VCC/3V3 and GND. All one-to-one.
The RX1/TX1/GND pins are clearly marked on the Due adapter, and the location of the VCC pins can be taken from any Arduino Due board specification (it's basically in the opposite corner of the GND pin):

![IMG_3811](https://github.com/user-attachments/assets/be688fcc-8944-40b3-b9ea-b27929b9253f)

You can use a tool like the one you posted, but you can also just use jumper wires to connect both ESP32 board and adapter.
On the ESP32, either use pins 16 for RX1 and 17 for TX1, or, if you use a non-30pin-compatible board, choose any free two pins you like and enter them in `bus_pins` in `BSB_LAN_config.h`.

**Comment by rroblik** [comment]
Wow so quick answer ! 
i have jumper wires already but will that need be soldered?

As I have not a precision tool for soldering...

**Reply by fredlcore** [reply]
You just need to make sure they connect somehow...

**Comment by rroblik** [comment]
Hi @fredlcore and happy new year !

I'm doing the move from the Due to the ESP32, but I'm encoutering some issue

Here is the telnet output : 
```
258916 Ping!
Found 1 devices in category list.
Scanning devices on the bus...
LAN->ALL  QINF      3D050064
DC C2 7F 0B 01 3D 05 00 64 37 B4
Bus devices found:
Device family: 0
Device variant: 0
 query failed
 query failed
 query failed
710
 query failed
 query failed
 query failed
8700
318918 Ping!
Found 1 devices in category list.
Scanning devices on the bus...
LAN->ALL  QINF      3D050064
DC C2 7F 0B 01 3D 05 00 64 37 B4
Bus devices found:
Device family: 0
Device variant: 0
```

Seems there is a communication error 😢 

I've just uploaded the latest program to the ESP, do the 4 wires connection as you said earlyer...
Maybe I missed something ? I wired the VCC of your board to the 3V3 pin on the ESP, is that correct ? 
I ask to myself if it's a wiring issue, software issue, hardware issue ...

Side note : I found the Due version of the ino file on my hardisk (aka : *my legacy*) and seems to be version 2.0.107 but no `BSB_LAN_custom_defs.h` file in this project 😞 
Accordling to the doc, when I do "device specific parameter list" on the web interface I have he following error :
`BSB-LAN cannot connect to the heating system. Please make sure that the adapter is connected to the heater and that the heater is turned on. Then restart BSB-LAN to connect to the heater.`

Thanks !

**Comment by fredlcore** [comment]
Ok, I'm sorry, above I wrote to connect to RX/TX on the ESP32, but it must be RX1/TX1 which are pins 16 and 17 on the ESP32 NodeMCU respectively (which I also wrote further down in that message). That's probably the issue.

**Comment by rroblik** [comment]
Oh ok I missed the 16/17 indication 🫣

just to be sure, on this picture it's GPIO16 and GPIO17 is that true?

![image](https://github.com/user-attachments/assets/23e1193b-9737-4763-9e5f-be32eefc70d3)

**Comment by fredlcore** [comment]
Im the end, the pin settings on BSB_LAN_config.h determine the pin used, irrespective of the label. It's strange to me that they have RX0/TX0 and RX2/TX2, but not RX1/RX2...

**Comment by rroblik** [comment]
Yes for me also that's why I ask you confirmation. I have a Wroom ESP32 maybe the pin out is not the same as what you think about on a nodeMCU. I'm a newbie on the topic so...

In the config file we just have to put the GPIO pins number it that true?

**Comment by fredlcore** [comment]
Yes

**Comment by rroblik** [comment]
Hello @fredlcore ; thanks all seems to work now !
I've jut emailed the device specific parameters list.

Thanks again !
